src/crimson/thread/ThreadPool.cc

```diff
@@ -0,0 +1,50 @@
+#include "ThreadPool.h"
+
+namespace ceph::thread {
+
+ThreadPool::ThreadPool(size_t n)
+  : threads{create_threads(n)}
+{}
+
+ThreadPool::~ThreadPool()
+{
+  stop = true;
+  cond.notify_all();
+  join();
```

Comment:
it looks like there could still be items in the queue here which need to be freed (and their associated promises failed with an exception?). as an alternative, maybe `loop()` should continue to process items until empty && stopping - this should work as long as `stop` is only set by the destructor, meaning that no other threads are calling into `submit()`

Reply:
fixed.